### PR TITLE
Fix failing tests due to notifications related to update in output

### LINKED ISSUE #6618: Tests might fail if update notification message is displayed in output
/kind bug
/area testing

Noticed this while investigating failures on #6500:

<details>
<summary>cmd_add_binding_test</summary>

```
Expected
      <string>: apiVersion: binding.operators.coreos.com/v1alpha1
      kind: ServiceBinding
      metadata:
        creationTimestamp: null
        name: aname
      spec:
        application:
          group: apps
          kind: Deployment
          name: app
          version: v1
        bindAsFiles: true
        detectBindingResources: true
        services:
        - group: postgresql.k8s.enterprisedb.io
          id: aname
          kind: Cluster
          name: cluster-sample
          resource: clusters
          version: v1
      status:
        secret: ""
      
      ---
      A newer version of odo (v3.7.0) is available,
      visit https://github.com/redhat-developer/odo/releases to update.
      If you wish to disable this notification, run:
      odo preference set UpdateNotification false
      ---
      
  to be equivalent to
      <string>: apiVersion: binding.operators.coreos.com/v1alpha1
      kind: ServiceBinding
      metadata:
        creationTimestamp: null
        name: aname
      spec:
        application:
          group: apps
          kind: Deployment
          name: app
          version: v1
        bindAsFiles: true
        detectBindingResources: true
        services:
        - group: postgresql.k8s.enterprisedb.io
          id: aname
          kind: Cluster
          name: cluster-sample
          resource: clusters
          version: v1
      status:
        secret: ""
      
  In [It] at: /go/odo_1/tests/integration/cmd_add_binding_test.go:53

...
Summarizing 2 Failures:
  [FAIL] odo add binding command tests [It] should create a binding using the workload parameter and naming strategy
  /go/odo_1/tests/integration/cmd_add_binding_test.go:86
  [FAIL] odo add binding command tests [It] should create a binding using the workload parameter
  /go/odo_1/tests/integration/cmd_add_binding_test.go:53

Ran 343 of 611 Specs in 1188.567 seconds
FAIL! -- 341 Passed | 2 Failed | 0 Pending | 268 Skipped
```

</details>

<details>
<summary>cmd_devfile_build_images_test</summary>

```
    [odo] build -t localhost:5000/devfile-nodejs-deploy:0.1.0 -f /tmp/221699204/Dockerfile /tmp/221699204
    [odo] 
 âœ“  Building image locally [5ms]
    [odo] 
    [odo] ---
    [odo] A newer version of odo (v3.7.0) is available,
    [odo] visit https://github.com/redhat-developer/odo/releases to update.
    [odo] If you wish to disable this notification, run:
    [odo] odo preference set UpdateNotification false
    [odo] ---
    Setting current dir to: /go/odo_1/tests/integration
    Deleting dir: /tmp/221699204
    Deleting dir: /tmp/3623255060
  << End Captured GinkgoWriter Output

  Expected
      <string>: odo preference set UpdateNotification false
  to be equivalent to
      <string>: build -t localhost:5000/devfile-nodejs-deploy:0.1.0 -f /tmp/221699204/Dockerfile /tmp/221699204
  In [It] at: /go/odo_1/tests/integration/cmd_devfile_build_images_test.go:125
...
Summarizing 4 Failures:
  [FAIL] odo devfile build-images command tests label nocluster when using a devfile.yaml containing an Image component with a build context [It] should build image via Docker if build context references PROJECT_SOURCE env var [nocluster]
  /go/odo_1/tests/integration/cmd_devfile_build_images_test.go:125
  [FAIL] odo devfile build-images command tests label nocluster when using a devfile.yaml containing an Image component with a build context [It] should build image via Podman if build context references PROJECT_SOURCE env var [nocluster]
  /go/odo_1/tests/integration/cmd_devfile_build_images_test.go:125
  [FAIL] odo devfile build-images command tests label nocluster when using a devfile.yaml containing an Image component with no build context [It] should build image via Podman by defaulting build context to devfile path [nocluster]
  /go/odo_1/tests/integration/cmd_devfile_build_images_test.go:169
  [FAIL] odo devfile build-images command tests label nocluster when using a devfile.yaml containing an Image component with no build context [It] should build image via Docker by defaulting build context to devfile path [nocluster]
  /go/odo_1/tests/integration/cmd_devfile_build_images_test.go:169

Ran 113 of 611 Specs in 79.630 seconds
FAIL! -- 109 Passed | 4 Failed | 0 Pending | 498 Skipped
```

</details>

## Any logs, error output, etc?

All the tests below seem to have failed for the same reason:

- https://s3.eu-de.cloud-object-storage.appdomain.cloud/odo-tests-openshift-logs/pr-6500-openshift-tests-4093.txt
- https://s3.eu-de.cloud-object-storage.appdomain.cloud/odo-tests-openshift-logs/pr-6500-nocluster-tests-1043.txt
- https://s3.eu-de.cloud-object-storage.appdomain.cloud/odo-tests-openshift-logs/pr-6500-kubernetes-tests-4057.txt
- https://s3.eu-de.cloud-object-storage.appdomain.cloud/odo-tests-openshift-logs/pr-6500-windows-tests-3292.txt

### PR DESCRIPTION
**Pull request by ritudes**
**What type of PR is this:**

/area testing

**What does this PR do / why we need it:**
->It fixes the issue of failing tests due to the display of update notifications in the output.

**Which issue(s) this PR fixes:**

Fixes #6618 

**PR acceptance criteria:**

- [ ] Unit test 

- [x] Integration test 

- [ ] Documentation 

**How to test changes / Special notes to the reviewer:**

### CODE REVIEW

**Review by rm3l — COMMENTED** [review]
@ritudes @anandrkskd LGTM to me, but besides this and to avoid similar issues in the future, can we also change the checks in those tests to make them less strict?
As discussed in https://github.com/redhat-developer/odo/issues/6618#issuecomment-1440244343, we should not expect an exact match, but it should be fine if we check that the output just contains the expected string.

**Comment by anandrkskd** [comment]
>  it should be fine if we check that the output just contains the expected string.
You mean to use the func `BeEquivalentTo` instead of what we are using right now in our checks ?

**Comment by rm3l** [comment]
> > it should be fine if we check that the output just contains the expected string.
> 
> You mean to use the func `BeEquivalentTo` instead of what we are using right now in our checks ?

The failing tests reported in https://github.com/redhat-developer/odo/issues/6618#issue-1595197802 are currently too strict, IMO, because they are using [`BeEquivalentTo`](https://onsi.github.io/gomega/#beequivalenttoexpected-interface) (which expects the actual value to be kind of equal to the expected value). I think we could be less strict by using `ContainSubstring` instead.

**Comment by anandrkskd** [comment]
> I think we could be less strict by using ContainSubstring instead.

understood, Thanks @rm3l 
will make updates

File: tests/integration/cmd_devfile_build_images_test.go [hunk 1]
```diff
@@ -124,7 +124,7 @@ var _ = Describe("odo devfile build-images command tests", func() {
 						containerImage := "localhost:5000/devfile-nodejs-deploy:0.1.0" // from Devfile yaml file
 						dockerfilePath := filepath.Join(commonVar.Context, "Dockerfile")
 						buildCtx := commonVar.Context
-						Expect(lines[nbLines-2]).To(BeEquivalentTo(
+						Expect(lines[nbLines-2]).To(ContainSubstring(
```
Comment (rm3l) [inline tests/integration/cmd_devfile_build_images_test.go:127]: I think checking an exact line number to match might be still a bit fragile, so I'd suggest checking that `stdout` contains the expected string:

```suggestion
						Expect(stdout).To(ContainSubstring(
```

File: tests/integration/cmd_devfile_build_images_test.go [hunk 2]
```diff
@@ -168,7 +168,7 @@ var _ = Describe("odo devfile build-images command tests", func() {
...
```
Comment (rm3l) [inline tests/integration/cmd_devfile_build_images_test.go:171]: I think checking an exact line number to match might be still a bit fragile, so I'd suggest checking that `stdout` contains the expected string:

```suggestion
						Expect(stdout).To(ContainSubstring(
```

**Review by rm3l — CHANGES_REQUESTED** [review]
A few comments on `cmd_devfile_build_images_test.go`

**Review by rm3l — APPROVED** [review]
Thanks for the changes, @anandrkskd and @ritudes!

**Comment by rm3l** [comment]
```
user-guides/quickstart/docs-mdx/dotnet/dotnet_odo_dev_output.mdx
  Expected
      <string>:   (
        	"""
        	... // 9 identical lines
        	â€¢  Waiting for Kubernetes resources  ...
        	âš   Pod is Pending
      - 	âœ“  Pod is Running
        	âœ“  Syncing files into the container [1s]
        	âœ“  Building your application in container (command: build) [1s]
        	... // 9 identical lines
        	[Ctrl+c] - Exit and delete resources from the cluster
        	[p] - Manually apply local changes to the application on the cluster
      + 	âœ“  Pod is Running
        	```
        	"""
        )
      
  to be empty
  In [It] at: /go/odo_1/tests/documentation/user-guides/doc_user_guides_quickstart_test.go:208
```

To be addressed by https://github.com/redhat-developer/odo/pull/6545

/override Kubernetes-Integration-Tests/Kubernetes-Docs-Integration-Tests

**Pull request merged**